### ISSUE TITLE
Variables that has dot can't be parsed

### ISSUE THREAD

**Issue by minh0722** [JamesOsgood/mongodb-grafana#15]
In my grafana I have a variable **win_version** that retrieve os version in my db and they have dots, i.e. windows 10 version 10.0.14393. When trying to match the os version with the variable in aggregate, JSON.parse() seems to fail because it seems the variable is not a string for some reason. The error is **Unexpected token . in JSON at position 103**

The variable's query is:
`db.session_info.aggregate([{"$group" : {"_id" : "$os_version"}}])`

Here's my query:
```
db.session_info.aggregate([
    {"$match" : {
        "time" : { "$gte" : "$from", "$lte" : "$to" },
        "os_version" : "$win_version"
    }},
    {"$bucketAuto" : {
        "groupBy" : "$time",
        "buckets" : "$dateBucketCount",
        "output" : { "count" : { "$sum" : 1 } }
    }},
    {"$project" : {
        "name" :  { "$concat" : ["Windows ", "$win_version"] },
        "value" : "$count",
        "ts" : "$_id.min",
        "_id" : 0
    }} 
])
```

**Comment by minh0722** [comment]
The problem might be that taking the value of grafana variables doesn't work correctly with the mongodb_grafana backend code

**Comment by JamesOsgood** [comment]
Can you share a sample document and I can test?

**Comment by minh0722** [comment]
```
{
    "_id" : "dd94aebf-8199-4664-be92-9c9a9b7be75e",
    "time" : ISODate("2018-07-06T07:14:45.000Z"),
    "os_version" : "10.0.16299"
}

{
    "_id" : "bee7d725-85c0-4804-879b-452edee5bef5",
    "time" : ISODate("2018-07-06T07:14:58.000Z"),
    "os_version" : "10.0.16299"
}

{
    "_id" : "d3c393c4-bc1c-4454-8e97-0a247d1cecc4",
    "time" : ISODate("2018-07-06T07:15:23.000Z"),
    "os_version" : "10.0.16299"
}

{
    "_id" : "c26ee12f-223c-4a9a-855b-ab93b4efdf12",
    "time" : ISODate("2018-07-06T07:15:44.000Z"),
    "os_version" : "10.0.15063"
}

{
    "_id" : "dd58ad36-2d1f-4322-bb0f-75732c2cd781",
    "time" : ISODate("2018-07-06T07:15:56.000Z"),
    "os_version" : "10.0.14393"
}
```

**Comment by JamesOsgood** [comment]
Hi,

I checked in a fix for this last night. The variables were being treated as regexes so that ended up with the . being escaped.

Should work now

James

**Comment by minh0722** [comment]
Sorry took me so long to start testing if this is fixed, but the variable is parsed as "{10.0.14393}". The curly braces is redundant and I couldnt match the value with it

**Comment by liyuanqiu** [comment]
> Sorry took me so long to start testing if this is fixed, but the variable is parsed as "{10.0.14393}". The curly braces is redundant and I couldnt match the value with it

I met this problem too. Have you found a solution?
